src/parser.y

```diff
@@ -710,6 +710,12 @@ Term '[' ']' '?' {
 Term '[' ']' %prec NONOPT {
   $$ = block_join($1, gen_op_simple(EACH));
 } |
+Term '.' '[' ']' '?' {
+  $$ = block_join($1, gen_op_simple(EACH_OPT));
+} |
+Term '.' '[' ']' %prec NONOPT {
+  $$ = block_join($1, gen_op_simple(EACH));
+} |
 Term '[' Exp ':' Exp ']' '?' {
   $$ = gen_slice_index($1, $3, $5, INDEX_OPT);
 } |
```

Comment:
@nicowilliams Sorry i should have marked this PR as draft as i think we probably want to support extra dot with all the different slice variants? also probably good with some basic tests for it. Then also thinking if there is some nicer way to write the rules that don't need as much duplication?

Reply:
We can always add that later.